ste/jobStatusManager.go

```diff
@@ -78,41 +80,70 @@ func (jm *jobMgr) handleStatusUpdateMessage() {
 
 	for {
 		select {
-		case msg := <-jstm.partCreated:
-			js.CompleteJobOrdered = js.CompleteJobOrdered || msg.IsFinalPart
-			js.TotalTransfers += msg.TotalTransfers
-			js.FileTransfers += msg.FileTransfers
-			js.FolderPropertyTransfers += msg.FolderTransfer
-			js.TotalBytesEnumerated += msg.TotalBytesEnumerated
-			js.TotalBytesExpected += msg.TotalBytesEnumerated
-
-		case msg := <-jstm.xferDone:
-			msg.Src = common.URLStringExtension(msg.Src).RedactSecretQueryParamForLogging()
-			msg.Dst = common.URLStringExtension(msg.Dst).RedactSecretQueryParamForLogging()
-
-			switch msg.TransferStatus {
-			case common.ETransferStatus.Success():
-				js.TransfersCompleted++
-				js.TotalBytesTransferred += msg.TransferSize
-			case common.ETransferStatus.Failed(),
-				common.ETransferStatus.TierAvailabilityCheckFailure(),
-				common.ETransferStatus.BlobTierFailure():
-				js.TransfersFailed++
-				js.FailedTransfers = append(js.FailedTransfers, msg)
-			case common.ETransferStatus.SkippedEntityAlreadyExists(),
-				common.ETransferStatus.SkippedBlobHasSnapshots():
-				js.TransfersSkipped++
-				js.SkippedTransfers = append(js.SkippedTransfers, msg)
+		// processMsg is high priority channel, used to process any messages lying on xferDone channel.
+		case <-jstm.processMsg:
+			jm.Log(pipeline.LogError, fmt.Sprintf("len: %v, cap: %v", len(jstm.xferDone), cap(jstm.xferDone)))
+			for msg := range jstm.xferDone {
+				msg.Src = common.URLStringExtension(msg.Src).RedactSecretQueryParamForLogging()
+				msg.Dst = common.URLStringExtension(msg.Dst).RedactSecretQueryParamForLogging()
+
+				switch msg.TransferStatus {
+				case common.ETransferStatus.Success():
+					js.TransfersCompleted++
+					js.TotalBytesTransferred += msg.TransferSize
+				case common.ETransferStatus.Failed(),
+					common.ETransferStatus.TierAvailabilityCheckFailure(),
+					common.ETransferStatus.BlobTierFailure():
+					js.TransfersFailed++
+					js.FailedTransfers = append(js.FailedTransfers, msg)
+				case common.ETransferStatus.SkippedEntityAlreadyExists(),
+					common.ETransferStatus.SkippedBlobHasSnapshots():
+					js.TransfersSkipped++
+					js.SkippedTransfers = append(js.SkippedTransfers, msg)
+				}
```

Comment:
let's move this to a new helper function handleXferDoneMsg() to avoid code duplication with the other case below.